### PR TITLE
home: fix & enhance documentation

### PR DESCRIPTION
**Pull request by utkarshgupta137**
### What does this PR try to resolve?

Fixes the documentation to specify that the `HOME` environment variable will be returned on Unix even if it is empty.
Also cleaned up & improved documentation in other areas.

### Additional information

Related: #12023

### CODE REVIEW

File: crates/home/src/lib.rs [hunk 1]
```diff
@@ -17,12 +8,16 @@
...
 //! `rustup_home`, which are the canonical way to determine the
-//! location that Cargo and rustup store their data.
+//! location that Cargo and rustup use to store their data.
+//! The `env` module contains utilities for mocking the process environment
+//! by Cargo and rustup.
 //!
 //! See also this [discussion].
 //!
+//! See also the [`dirs`](https://docs.rs/dirs) crate.
+//!
```
Comment (weihanglo) [inline crates/home/src/lib.rs:20]: The Cargo project usually avoids mentioning a specific crate in its documentation. I would recommend removing these.

**Review by weihanglo — COMMENTED** [review]
Thanks! Looks good.

File: crates/home/src/lib.rs (hunk 1 above)
Comment (utkarshgupta137) [reply crates/home/src/lib.rs]: Done.

Comment (weihanglo) [reply crates/home/src/lib.rs]: Thanks. Could you also bump the crate version to 0.5.6? We might not make a release right away though.

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thank you!

@bors r+

**Comment by utkarshgupta137** [comment]
One thing I forgot to mention:
- For Unix, if the `HOME` variable is empty, then an empty path is returned.
- For Windows, if the `USERPROFILE` is empty, then it still falls back to the OS API.

I personally think this crate should be consistent regarding this. My PR #12023 was about fixing the behavior for Unix but was closed on the grounds that this is what Go & Python do. I think that on the same basis, if the `USERPROFILE` variable is empty, an empty path should be returned on Windows too.

**Pull request merged**